containers/ecr-viewer/src/app/services/userService.ts

```diff
@@ -204,6 +204,47 @@ const createUserQuery = async (
   return uuid;
 };
 
+/**
+ *
+ * @param db the database connection
+ * @param email email of the user to create or update (case-insensitive)
+ * @param uuid uuid of the user to create or update
+ * @returns whether an inactive user with the same email exists
+ * @throws UserFacingError when an active user with the same email exists
+ */
+const checkDupeEmail = async (
+  db: Transaction<Core>,
+  email: string | null | undefined,
+  uuid: string,
+): Promise<boolean> => {
+  if (!email) {
+    return false;
+  }
+
+  const user = await db
+    .selectFrom("user")
+    .selectAll()
+    .where((eb) =>
+      eb(eb.fn<string>("LOWER", [eb.ref("email")]), "=", email?.toLowerCase()),
+    )
+    .where("uuid", "!=", uuid)
+    .executeTakeFirst();
+
+  // inactive user does not exist
+  if (!user) {
+    return false;
+  }
+
+  if (user.status === "active") {
```

Comment:
we should probably always throw here and not just when active to avoid a strange inconsistency in the database state. Maybe a distinct error message though?